samples/net/openthread/coap_server/README.rst

```diff
@@ -1,13 +1,14 @@
 .. _coap_server_sample:
 
-openthread: coap_server
-##################
+OpenThread: CoAP Server
+#######################
 
-The openthread coap_server sample demonstrates interactions between nodes performing different Thread roles with the use of OpenThread and CoAP protocol.
+The OpenThread CoAP Server sample demonstrates interactions between nodes performing different Thread roles with the use of OpenThread and CoAP protocol.
```

Comment:
So this is basically identical to the Client sample? Can I use both interchangeably? ;)
Please expand this introduction a bit and make it meaningful for both the client and the server as if they were standalone samples, for example, in very simplified manner:
```
The OpenThread CoAP client sample demonstrates controlling light resources of other nodes within an OpenThread network. To show this interaction, the sample requires another sample with a light resource available, for example <link to CoAP server>.
```  
And similar description for the CoAP server. Very basic, as you just want to understand what this sample is about when reading this.